diff --git a/src/aws-cpp-sdk-core/source/auth/CrtCredentialsProvider.cpp b/src/aws-cpp-sdk-core/source/auth/CrtCredentialsProvider.cpp
index f58deb94c41a..5bdbe83f3e59 100644
--- a/src/aws-cpp-sdk-core/source/auth/CrtCredentialsProvider.cpp
+++ b/src/aws-cpp-sdk-core/source/auth/CrtCredentialsProvider.cpp
@@ -99,8 +99,8 @@ AWSCredentials CrtCredentialsProvider::ExtractCredentialsFromCrt(const Aws::Crt:
   credentials.SetAWSAccessKeyId({reinterpret_cast<char*>(accountIdCursor.ptr), accountIdCursor.len});
   const auto secretKeyCursor = crtCredentials.GetSecretAccessKey();
   credentials.SetAWSSecretKey({reinterpret_cast<char*>(secretKeyCursor.ptr), secretKeyCursor.len});
-  const auto expiration = crtCredentials.GetExpirationTimepointInSeconds();
-  credentials.SetExpiration(DateTime{static_cast<double>(expiration)});
+  const uint64_t expiration = crtCredentials.GetExpirationTimepointInSeconds();
+  credentials.SetExpiration(DateTime{expiration});
   const auto sessionTokenCursor = crtCredentials.GetSessionToken();
   credentials.SetSessionToken({reinterpret_cast<char*>(sessionTokenCursor.ptr), sessionTokenCursor.len});
   return credentials;
diff --git a/src/aws-cpp-sdk-core/source/utils/DateTimeCommon.cpp b/src/aws-cpp-sdk-core/source/utils/DateTimeCommon.cpp
index e50c32d2539e..937a4ec3820a 100644
--- a/src/aws-cpp-sdk-core/source/utils/DateTimeCommon.cpp
+++ b/src/aws-cpp-sdk-core/source/utils/DateTimeCommon.cpp
@@ -1142,8 +1142,17 @@ DateTime::DateTime(double secondsSinceEpoch) : m_valid(true)
 
 DateTime::DateTime(uint64_t secondsSinceEpoch) : m_valid(true)
 {
-  std::chrono::duration<uint64_t, std::chrono::seconds::period> timestamp(secondsSinceEpoch);
-  m_time = std::chrono::system_clock::time_point(timestamp);
+    const auto maxTime = (std::chrono::system_clock::time_point::max)();
+    const auto maxSeconds = std::chrono::duration_cast<std::chrono::seconds>(maxTime.time_since_epoch()).count();
+
+    if (secondsSinceEpoch > static_cast<uint64_t>(maxSeconds))
+    {
+        m_time = maxTime;
+    }
+    else
+    {
+        m_time = std::chrono::system_clock::time_point(std::chrono::seconds(static_cast<std::chrono::seconds::rep>(secondsSinceEpoch)));
+    }
 }
 
 DateTime::DateTime(const Aws::String& timestamp, DateFormat format) : m_valid(true)
